### DISCUSSION TITLE
Potential Issues with Immersion freezing in hetfrz.F90

### DISCUSSION THREAD

**Discussion by cbardeen** [NCAR/amwg_dev#199] [TF3:Time-stepping, subcycling]
This was a discussion that started in e-mail and Andrew asked that a github discussion be created for it. So I have copied the e-mail that I sent out. These solutions have not been tested as the immersion freezing is not currently being used, but those routines are being ported to the CARMA cloud model so Yunqian and I discussed some of the potential issues. Francis is also working on refactoring these routines so that they will work with the CARMA aerosols model that Simone is working on.

----------

I was talking with Yunqian yesterday about the immersion freezing code as she is trying to copy the same logic into CARMA. I came across a couple of things that would seem to make nucleation too active. I think issue #2 is a bug and should be fixed.

I am not sure how much of an effect either of these will have. For TTL cirrus, issue #2 made a big difference as that is a region with mostly sulfates in coarse mode. For immersion freezing, you are in a mixed phase region, so the fraction of the aerosol mass in sulfates is probably smaller than in the TTL, but it might still be significant. If someone has a run handy with those fields, they could look at the mass fraction and see if it looks like it would be a big deal. The simple work around for Issue #1 could be tried if someone is interested, just to see if it looks like it has any effect on the overall nucleation rate. If it is a big effect, then perhaps splitting cloudborne into ice and liquid is worth the effort.

Thanks,

Chuck

Issue #1

One of the things I am worried about is whether the aerosol will be "used up" in the process. The nucleation rate is a function of the cloudborne aerosol number for different species and modes:

   if (.not.tot_in) then
      if (do_bc) frzbcimm = frzbcimm+MIN(limfacbc*total_cloudborne_aer_num(id_bc)/deltat, &
                            total_cloudborne_aer_num(id_bc)/deltat*(1._r8-exp(-Jimm_bc*deltat))) 

      if (.not. pdf_imm_in) then
         if (do_dst1) frzduimm = frzduimm+MIN(1*total_cloudborne_aer_num(id_dst1)/deltat, & 
                                 total_cloudborne_aer_num(id_dst1)/deltat*(1._r8-exp(-Jimm_dust_a1*deltat)))
         if (do_dst3) frzduimm = frzduimm+MIN(1*total_cloudborne_aer_num(id_dst3)/deltat, &
                                 total_cloudborne_aer_num(id_dst3)/deltat*(1._r8-exp(-Jimm_dust_a3*deltat)))
      else
         if (do_dst1) frzduimm = frzduimm+MIN(1*total_cloudborne_aer_num(id_dst1)/deltat,        &
                                 total_cloudborne_aer_num(id_dst1)/deltat*(1._r8-sum_imm_dust_a1))
         if (do_dst3) frzduimm = frzduimm+MIN(1*total_cloudborne_aer_num(id_dst3)/deltat,        &
                                 total_cloudborne_aer_num(id_dst3)/deltat*(1._r8-sum_imm_dust_a3))
      end if

   else
      if (do_bc) frzbcimm = frzbcimm+MIN(limfacbc*fn(id_bc)*total_aer_num(id_bc)/deltat, & 
                            fn(id_bc)*total_aer_num(id_bc)/deltat*(1._r8-exp(-Jimm_bc*deltat))) 

      if (.not. pdf_imm_in) then
         if (do_dst1) frzduimm = frzduimm+MIN(1*fn(id_dst1)*total_aer_num(id_dst1)/deltat, &
                                 fn(id_dst1)*total_aer_num(id_dst1)/deltat*(1._r8-exp(-Jimm_dust_a1*deltat)))
         if (do_dst3) frzduimm = frzduimm+MIN(1*fn(id_dst3)*total_aer_num(id_dst3)/deltat, &
                                 fn(id_dst3)*total_aer_num(id_dst3)/deltat*(1._r8-exp(-Jimm_dust_a3*deltat)))
      else
         if (do_dst1) frzduimm = frzduimm+MIN(1*fn(id_dst1)*total_aer_num(id_dst1)/deltat,        &
                                 fn(id_dst1)*total_aer_num(id_dst1)/deltat*(1._r8-sum_imm_dust_a1))
         if (do_dst3) frzduimm = frzduimm+MIN(1*fn(id_dst3)*total_aer_num(id_dst3)/deltat,        &
                                 fn(id_dst3)*total_aer_num(id_dst3)/deltat*(1._r8-sum_imm_dust_a3))
      end if
   end if

Physically, when a liquid drop with an aerosol in it freezes, then the aerosol is no longer in a cloud drop it is in an ice crystal. Unfortunately in CAM, we do not advect cloudborne aerosol nor do we have seperate varaibles for cloudborne aerosol in liquid and in ice. Thus immersion freezing thinks that all aerosol that should be in the ice are in the liquid and as nucleation progresses there is no reduction in the number of cloudborne aerosols in the liquid. The most precise way to fix this would be to duplicate the set of cloudborne aerosol variables and have one set for liquid and another set for ice. Then when immersion nucleation happens it would shift the aerosols from the liquid variables to the ice variables. You would also need to move the cloudborne aerosols when clouds freeze and when they melt, ... So that could be a fair amount of work to make sure you have it all correct (revisiting ndrop, ...). A simplier solution that might have some effect in limiting the immersion nucleation would be to use a cloud mass fraction to determine the amount of the aerosol that is in the liquid and ice clouds:

num_cloudborne_ice = ice_mass / (ice_mass + liquid_mass) * num_cloudborne
num_cloudborne_liq = liq_mass / (ice_mass + liquid_mass) * num_cloudborne

This would at least have the property that as more cloud is converted to ice, the available nuclei for immersion freezing would go down, since they should only come from num_cloudborne_liq.

Issue #2

Also, in looking at how the cloudborne aerosol number is calculated, it seems like there is a problem related to the mass fractions in the coarse mode. This seems like the same problem that was originally in nucleate_ice where they have forgotten about the sulfates in the coarse mode. In get_aer_num, you have:

      dmc_imm = aer_cb(ii,kk,dst_coarse)
      ssmc_imm = aer_cb(ii,kk,ncl_coarse)

      if (dmc_imm > 0._r8) then
         dst3_num_imm = dmc_imm/(ssmc_imm+dmc_imm) * aer_cb(ii,kk,num_coarse)*1.0e-6_r8 ! #/cm^3
      else
         dst3_num_imm = 0.0_r8
      end if

This would make it seem like you have a lot of cloudborne dust in places were you really have a lot of sulfate and should make the immersion nucleation too active. I think it should be:

      dmc_imm = aer_cb(ii,kk,dst_coarse)
      ssmc_imm = aer_cb(ii,kk,ncl_coarse)
      so4_imm = aer_cb(ii,kk,so4_coarse)

      if (dmc_imm > 0._r8) then
         dst3_num_imm = dmc_imm/(ssmc_imm+dmc_imm+so4_imm) * aer_cb(ii,kk,num_coarse)*1.0e-6_r8 ! #/cm^3
      else
         dst3_num_imm = 0.0_r8
      end if

It is possible that they are trying to assume that the coarse mode is all coated dust and sea salt in the mixed phase region and that there is no stand-alone sulfate; however, this assumption would not work well in regions where there are externally mixed sulfate particles.

**Comment by adamrher** [comment]
> These solutions have not been tested as the immersion freezing is not currently being used

It is my understanding the cam6 microphysics uses this code, but that the latest version of PUMAS reverts the immersion freezing to the Bigg's scheme, which doesn't consider the impact of aerosols on ice nucleation. Is that what you mean?

> Thus immersion freezing thinks that all aerosol that should be in the ice are in the liquid and as nucleation progresses there is no reduction in the number of cloudborne aerosols in the liquid.

I'm a little surprised at the lack of accounting here. Is this a bug or a potentially invalid assumption (that there is always plenty of liquid?)

Thanks for the proposed solutions. I think the more involved solution (create separate liquid and ice cloudborne number vars) is possible given it sounds like we are 6-months out from calibrating PUMAS in CAM7 (based on notes from the Friday mtg).

**Reply by adamrher** [reply]
Regarding the bug, number 2. Makes sense to me. The hard coding of species seems the real issue here, but I'm just the peanut gallery.

**Comment by andrewgettelman** [comment]
Thanks @cbardeen and @adamrher. I think we (that's the royal we, probably Chuck or Simone) should see about implementing these proposed adjustments and we can see what it does. 

What is Simone's github username? We can flag her in the discussion. 

@adamrher, yes, what I did for the current cam_dev is to turn off the CNT for immersion freezing that uses aerosols so that part is not aerosol aware (back to just an empirical temperature dependent fit). But we should see if we can bound the immersion freezing this way.

I'm happy to try to test if Chuck wants to take a cut at the code mods.

**Reply by jedwards4b** [reply]
https://github.com/tilmes ^

**Reply by tilmes** [reply]
my user name is tilmes

**Comment by andrewgettelman** [comment]
@cbardeen, one more comment:  on your number 2, is there much SO4 in the coarse mode (in the troposphere?). I guess we could try it. And for number 1, I think this is also worth trying to limit the immersion freezing and take out what is presumably already frozen.

Volunteers? This doesn't sound too hard to do.

By the way. If you look at the latest E3SM code, they have the same problems:
https://github.com/E3SM-Project/E3SM/blob/v2.0.0/components/eam/src/physics/cam/hetfrz_classnuc.F90
https://github.com/E3SM-Project/E3SM/blob/v2.0.0/components/eam/src/physics/cam/hetfrz_classnuc_cam.F90

I'm going to talk with them about it this week. Maybe I can get someone at PNNL to test it....

A

**Comment by tilmes** [comment]
@cbardeen @andrewgettelman @fvitt 
A question that came up when Francis and I talked about the code and we hope one of you can answer is:
The subroutine: hetfrz_classnuc_cam_save_cbaero  in hetfrz_classnuc_cam.F90 which get the cloud borne aerosol number and mmr 
<img width="634" alt="image" src="https://user-images.githubusercontent.com/22222754/202290479-a93931f7-80ab-4f4c-a0b5-e7a77f1b5001.png">

is called in microp_aero.F90 before ndrop (dropmixnuc) and nucleate_ice is called. So cloud borne aerosols in the heterogeneous freezing routine are not actually reflecting the cloud-borne aerosols that are assumed in nucleate ice. 

This variable "aer_cb" is then used in get_aer_num to calculate the dust number for the immersion freezing and with that the "total_cloudborne_aer_num". On the other hand the interstitial aerosol (to calculate the mass) is taking from the interstitial aerosols that exist after ndrop and nucleate ice.

Our question therefore, what is the reason for this inconsistency between cloud borne and interstitial, and wondered if that is another bug in the model???

**Comment by tilmes** [comment]
@cbardeen quick question, is alnsg_mode_accum the mean radius of the mode in this routine?

**Comment by cbardeen** [comment]
I suspect this is a bug. Initially nucleate_ice did not affect cloudborne aerosol; however, ndrop always has. It is odd the way this routine determines cloudborne, and it would seem to me that it should refelct the actual sstate at the time of the call, but you should ask the people who implented it if there is a reason they did it this way.

> On Nov 16, 2022, at 1:58 PM, tilmes ***@***.***> wrote:
> 
> 
> @cbardeen <https://github.com/cbardeen> @andrewgettelman <https://github.com/andrewgettelman> @fvitt <https://github.com/fvitt>
> A question that came up when Francis and I talked about the code and we hope one of you can answer is:
> The subroutine: hetfrz_classnuc_cam_save_cbaero in hetfrz_classnuc_cam.F90 which get the cloud borne aerosol number and mmr
>  <https://user-images.githubusercontent.com/22222754/202290479-a93931f7-80ab-4f4c-a0b5-e7a77f1b5001.png>
> is called in microp_aero.F90 before ndrop (dropmixnuc) and nucleate_ice is called. So cloud borne aerosols in the heterogeneous freezing routine are not actually reflecting the cloud-borne aerosols that are assumed in nucleate ice.
> 
> This variable "aer_cb" is then used in get_aer_num to calculate the dust number for the immersion freezing and with that the "total_cloudborne_aer_num". On the other hand the interstitial aerosol (to calculate the mass) is taking from the interstitial aerosols that exist after ndrop and nucleate ice.
> 
> Our question therefore, what is the reason for this inconsistency between cloud borne and interstitial, and wondered if that is another bug in the model???
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/NCAR/amwg_dev/discussions/199#discussioncomment-4161151>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AFLRNRG6JHPHLPFZCVRX7KTWIVDGLANCNFSM6AAAAAARZVBXUQ>.
> You are receiving this because you were mentioned.
> 

--------------------------------------------------------------
Charles Bardeen
National Center for Atmospheric Research
P.O. Box 3000, Boulder, CO 80307-3000
Phone: (303) 497-1752,  Fax: (303) 497-1400
***@***.***

**Comment by cbardeen** [comment]
I agree that #2 may not matter in this case. We could try #1. Is there a code base that you want to work with to try a test? If you point me to it, I can try to make the code change.

> On Nov 11, 2022, at 3:21 PM, Andrew Gettelman ***@***.***> wrote:
> 
> 
> @cbardeen <https://github.com/cbardeen>, one more comment: on your number 2, is there much SO4 in the coarse mode (in the troposphere?). I guess we could try it. And for number 1, I think this is also worth trying to limit the immersion freezing and take out what is presumably already frozen.
> 
> Volunteers? This doesn't sound too hard to do.
> 
> By the way. If you look at the latest E3SM code, they have the same problems:
> https://github.com/E3SM-Project/E3SM/blob/v2.0.0/components/eam/src/physics/cam/hetfrz_classnuc.F90 <https://github.com/E3SM-Project/E3SM/blob/v2.0.0/components/eam/src/physics/cam/hetfrz_classnuc.F90>
> https://github.com/E3SM-Project/E3SM/blob/v2.0.0/components/eam/src/physics/cam/hetfrz_classnuc_cam.F90 <https://github.com/E3SM-Project/E3SM/blob/v2.0.0/components/eam/src/physics/cam/hetfrz_classnuc_cam.F90>
> I'm going to talk with them about it this week. Maybe I can get someone at PNNL to test it....
> 
> A
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/NCAR/amwg_dev/discussions/199#discussioncomment-4120834>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AFLRNRFWY4TJFLGPFZHXBPDWH3BFJANCNFSM6AAAAAARZVBXUQ>.
> You are receiving this because you were mentioned.
> 

--------------------------------------------------------------
Charles Bardeen
National Center for Atmospheric Research
P.O. Box 3000, Boulder, CO 80307-3000
Phone: (303) 497-1752,  Fax: (303) 497-1400
***@***.***

**Reply by andrewgettelman** [reply]
Thanks @cbardeen for trying this. I would say to try with the latest code (cam6_3_084). But we should use the -phys cam_dev option (code I think is the same), and will need to turn on the CNT immersion freezing, which I switched off for cam_dev.  Hopefully that makes sense?

**Comment by fvitt** [comment]
Any ideas why the calculation of the accumulation mode number densities is different than the coarse mode number densities?  The calculation of the coarse mode num densities use mode number mixing ratio times the fraction of the species in the mode, which makes sense to me.  While the accum mode densities are calculated using the mass mixing ratio of the species multiplied by some num_to_mass conversion factors, which are hard wired in the code.  Is the some reason these 2 modes are treated differently?

**Comment by tilmes** [comment]
I setup a control and a first test case to do several test cases:
Control: current cam_dev pumas scheme (with the old immersion freezing routine)
Casedir: /glade/p/cesm/chwg_dev/tilmes/cases/cesm2_ch/f.e22.Fnudged.f09_f09_mg17.cam6_3_082.2018.001
Archive: /glade/scratch/tilmes/archive/f.e22.Fnudged.f09_f09_mg17.cam6_3_082.2018.001
aircraft flight track file: *hs* in atm/hist

Test1: Immersion freezing as in CAM6
Casedir: /glade/p/cesm/chwg_dev/tilmes/cases/cesm2_ch/f.e22.Fnudged.f09_f09_mg17.cam6_3_082.2018.immfrz.001
Archive: /glade/scratch/tilmes/archive/f.e22.Fnudged.f09_f09_mg17.cam6_3_082.2018.immfrz.001

Comparisons show that the test1 case (left) shows more ice number than the control case (as expected):
<img width="1485" alt="image" src="https://user-images.githubusercontent.com/22222754/206071174-e02ad207-cdce-4f3c-8ba1-343b7607a164.png">

Next I would like to do 2 tests, one that read the cloud borne aerosols before heterogeneous freezing and not before ndrop, and second, to include the fix by @cbardeen to move dust from cloud born into ice (Issue #1).

**Reply by andrewgettelman** [reply]
Good. That is exactly what I would expect to happen. That diagnostic (NUMICE) shows the issue pretty clearly. Hopefully the other tests can fix this....

**Reply by adamrher** [reply]
@andrewgettelman can you confirm what, if any science changes are coming in through this PUMAS DDT PR: https://github.com/ESCOMP/CAM/pull/632 ?

Simone and Chuck can I request that you add your initials as comments around your SourceMods going forward? I'd like to be able to see what SourceMods are responsible for the differences in these runs, and I'm having trouble seeing exactly what you did in test1 to revert to CAM6 immersion freezing in ```micro_pumas_v1.F90```. I agree with the approach of *not* setting -phys cam6 to revert to CAM6 immersion freezing as that changes all sorts of things and will make it impossible to pin down the cause of the differences.

**Reply by andrewgettelman** [reply]
Hi @adamrher. 

1. No science changes in the PUMAS DDT PR. Just cleaning up the calling tree to make development and CCPP porting easier. 
2. Yes, the approach here is to move forward with cam_dev, looking only at bringing back the CAM6 immersion freezing. The logic here was not incredibly easy to follow since we had to move some stuff around to minimize code changes. Happy to explain it if you want.

**Reply by tilmes** [reply]
@adamrher The part below is what has changed:
1. since use_hetfrz_classnuc is set to true, we move the code that is used in the control case above (immersion_freezing and contact_freezing) back into the block for "not. use_hetfrz_classnuc" because we are not using it in test1. Instead, in test1 we are setting the immersion freezing rates to what has been calculated in hetfrz_classnuc_cam.F90. This is how it was in CAM6.

<img width="798" alt="image" src="https://user-images.githubusercontent.com/22222754/206457427-bb7de612-ecc5-42be-b550-569b425d4b78.png">

**Reply by tilmes** [reply]
Here are some more results:  Control (black), Test1 (red):
<img width="640" alt="image" src="https://user-images.githubusercontent.com/22222754/206459154-3759afe0-492a-486c-b0c5-7d6889479e86.png">

**Comment by tilmes** [comment]
Test2: Immersion freezing as in CAM6, plus get cloud-borne aerosols in hetfrz_classnuc_cam.F90 same as interstitial aerosols. Remove call to get aer_cb in microp_aero.F90
Casedir: /glade/p/cesm/chwg_dev/tilmes/cases/cesm2_ch/f.e22.Fnudged.f09_f09_mg17.cam6_3_082.2018.immfrz.002
Archive: /glade/scratch/tilmes/archive/f.e22.Fnudged.f09_f09_mg17.cam6_3_082.2018.immfrz.002

**Reply by tilmes** [reply]
Here is Control (black) vs Test2 (red): Some improvement, reduction in NUM_ICE with this change:
<img width="647" alt="image" src="https://user-images.githubusercontent.com/22222754/206459574-fb43389e-07ce-4c9c-994f-01d67a750ad9.png">

**Reply by tilmes** [reply]
I just  confirmed with Xiaohong, that we should definitely implement this fix, it does not make sense to use cloud-borne aerosols from before ndrop.

**Reply by adamrher** [reply]
Isn't the red line still out to lunch in test2? The goal is try to reduce it down to somewhere near the black solution cntl, right?

**Reply by tilmes** [reply]
@adamrher The Test5 (next test I am just setting up) will include test2, and test4 changes (still waiting for test3 to get setup so leaving it out for now), plus I am adding the Issue#2 changes (as discussed above). Please note, tests2-4 just changed a single thing each to see what they do. The combined tests may be more effective plus there will be more changes, for example running with MAM5 will reduce dust, and running with the new SOA scheme for simple chemistry will reduce BC. All of this will push the values down eventually. We have to just see by how much.

**Reply by adamrher** [reply]
Will be curious to see the combined tests.

**Comment by tilmes** [comment]
@cbardeen For Test3, when you  have a chance, could you help me with the code that will move the aerosols from liquid to ice?

**Reply by tilmes** [reply]
@adamrher This is expected to maybe have the largest impact on the ice-number..

**Comment by tilmes** [comment]
Test4: Setting num_to_mass_in to false, but else it is like Test 1:
Casedir: /glade/p/cesm/chwg_dev/tilmes/cases/cesm2_ch/f.e22.Fnudged.f09_f09_mg17.cam6_3_082.2018.immfrz.004
Archive: /glade/scratch/tilmes/archive/f.e22.Fnudged.f09_f09_mg17.cam6_3_082.2018.immfrz.004

<img width="732" alt="image" src="https://user-images.githubusercontent.com/22222754/206288315-06e45ee2-d24d-4ffa-8a0e-15c0d00c7cec.png">

**Reply by tilmes** [reply]
Control (black), Test4 (red):
Quite a bit of a reduction in NUM_ICE in this test, we are now at 9.e0+5 (test4) compared to 1e+05 (control) while test1 is 2.4e+06.
<img width="636" alt="image" src="https://user-images.githubusercontent.com/22222754/206460175-6c52da1d-a246-4084-a8bb-7ba15fa0f53f.png">

**Reply by tilmes** [reply]
I also confirmed with Xiaohong that this is an OK change.

**Comment by tilmes** [comment]
Test5: Combine test2, test4, and fix listed in Issue#2 above
Casedir: /glade/p/cesm/chwg_dev/tilmes/cases/cesm2_ch/f.e22.Fnudged.f09_f09_mg17.cam6_3_082.2018.immfrz.005
Archive: /glade/scratch/tilmes/archive/f.e22.Fnudged.f09_f09_mg17.cam6_3_082.2018.immfrz.005

**Reply by tilmes** [reply]
The combined test did not reduce the number more, so a little disappointing, but I will double-check the test, and also do a MAM5 test... 

<img width="903" alt="image" src="https://user-images.githubusercontent.com/22222754/206561071-c12ae039-2410-41e3-8111-fceeb524b634.png">

**Comment by tilmes** [comment]
Test 6: 
Same as Test5 but adding MAM5 (reduced dust in the atmosphere)
Casedir: /glade/p/cesm/chwg_dev/tilmes/cases/cesm2_ch/f.e22.Fnudged.f09_f09_mg17.cam6_3_082.2018.immfrz.006
Archive: /glade/scratch/tilmes/archive/f.e22.Fnudged.f09_f09_mg17.cam6_3_082.2018.immfrz.006

**Comment by tilmes** [comment]
Test 7:
Casedir: /glade/p/cesm/chwg_dev/tilmes/cases/cesm2_ch/f.e22.Fnudged.f09_f09_mg17.cam6_3_082.2018.immfrz.007
Archive: /glade/scratch/tilmes/archive/f.e22.Fnudged.f09_f09_mg17.cam6_3_082.2018.immfrz.007

Same as Test 5 but with MAM5 (reduced dust in the atmosphere) and adding the new simple SOA description to CAM which results in reduced black carbon. This is about the maximum I see we can do, without the fix that Chuck  is going to implement (which is expected to have a large impact). We are getting somewhat closer to the old approach. 
<img width="659" alt="image" src="https://user-images.githubusercontent.com/22222754/206784404-b512388b-3f04-486f-8da5-175171f78849.png">
@andrewgettelman would you be willing to do some comparisons with the observations? The flight track output is in the archive directory.

**Reply by tilmes** [reply]
P.S.this is the same plot but for May 
<img width="663" alt="image" src="https://user-images.githubusercontent.com/22222754/206785416-443c1c5d-6348-46d8-90c5-d8ecf032b9da.png">
and a time evolution, for the tropics (left is the control run, right is Test 7)
<img width="1485" alt="image" src="https://user-images.githubusercontent.com/22222754/206786023-0b1e5751-645c-43b5-90ba-7255f4efe63a.png">

**Comment by adamrher** [comment]
Here's a zonal mean plot in linear (not log) color scale, for cntl (top left) and tests 001, 002, 004, 005, 007. This is an annual mean over a single year.

![temp_dzonal_NUMICE](https://user-images.githubusercontent.com/29961476/206790621-d3bbe6a7-3863-4e91-a169-29cd99302afe.png)

The mods in Test 004 clearly reduce NUMICE the most. What is the reasoning for executing this ```if (.not. num_to_mass_in) then``` block in test 004?

**Reply by tilmes** [reply]
@adamrher thanks for the plots. Using num_to_mass_in is using a calculation that is based on measurements (at a specific location and time) while if this is turned off, the number faction of an aerosol is actually calculated by the composition in the model at every grid point, so to me that is more physical. Xiaohong confirmed that if this gives better results we should turn this off.

**Comment by andrewgettelman** [comment]
I'm separately talking with Zachary McGraw who is looking at some of these modifications. He is suggesting that we could also reduce the dust used in the CNT immersion freezing by a factor to get the ice nucleation to be more reasonable. I think we might do that if this does not work. I might do it anyway as a test, depending on whether this last test works.

**Reply by tilmes** [reply]
Dust will be reduce when we add MAM5 (Test 7). We probably want to decide what to do, after adding in the changes in Test 3 (Issue #1).

**Comment by andrewgettelman** [comment]
That makes sense. I'll see where Zachary put in the dust fraction if we need that after test #3

**Reply by tilmes** [reply]
@andrewgettelman would you be able to do some comparisons with the flight track information?
It would be probably good to check where we are in comparison to those as well?
Thanks Simone

**Comment by andrewgettelman** [comment]
I think the problem with flight tracks is that the SOCRATES tracks are in regions with very little dust: so the impacts may not be seen there. 

I got code mods from Zachary to limit dust fraction in hetfrz_classnuc.F90, the same way we are limiting soot fraction (limfacbc): basically I've added an analogous limfacdst and set it to 0.1

I'm setting up a test run of that now, and will let you know how it goes. If test #3 does not work, this is a backup: it is consistent with other code and still has CNT immersion freezing back on.

**Comment by adamrher** [comment]
I suspect that the time-step sensitivity --- decreasing ice number with increasing mg_num_steps --- is related to something like what Simone is proposing in test#3, and won't be resolved by what we've tested so far. I'm requesting that if we do get a fix for test 003, that we run an F-case for a few different values of mg_num_steps ([similar to these experiments from Andrew](https://github.com/NCAR/amwg_dev/discussions/86#discussioncomment-2576855))

**Comment by andrewgettelman** [comment]
I'm going to add @zmcgraw to this thread so he can follow what is happening on the aerosol side of things.

**Comment by andrewgettelman** [comment]
Adding @zmcgraw1 to the thread. Correct github handle...

**Comment by andrewgettelman** [comment]
Following suggestions from @zmcgraw1, I've done a test where we limit the fraction of dust used for immersion freezing to 10% or 5%. This is an effective way at limiting the impact of the CNT code, and is consistent with what is already done for bc in hetfrz_classnuc.F90.  I will probably update the PUMAS submitted manuscript in review to use this change rather than go back to Bigg. With the changes you are making, I think this will ameleriorate the issue. 

Let me know how the last test pans out. I can include these changes in the paper if we want and add @cbardeen and @tilmes to it, or that can go in a further paper.

**Reply by tilmes** [reply]
@andrewgettelman yes, that is possible, however, what I am struggling with is, what are we aiming for as a goal? As you can see, adding MAM5, or the SOA approach will reduce dust and bc, also Chuck's approach may still reduce the ice number a bit. I am a little bit in the dark on what the goal is so just adding some factor does only make sense once we have the final configuration, correct? I would definelty suggest to all all the changes I made up to test 4 in the new Pumas case and then wait for Chuck's changes. Btw, I just did some CARMA runs using the old Meyer approach, leading to much smaller high-cloud forcing. What high-cloud values are we aiming for? And yes, I agree to update the Pumas paper, and would be happy to be part of it :)

**Reply by andrewgettelman** [reply]
The goal @tilmes I think is that we need to have a decent cloud and aerosol simulation. So it may be that just not all the dust should be participating in immersion freezing, as we have made this assumption for BC already. I'm fine with combining these. Just waiting on Chuck's tests and then I can do that.

Does that make sense?

**Comment by andrewgettelman** [comment]
@tilmes Happy new year!  Any updates on the last set of tests (test #3). I'm happy to try to do some quick climatological analysis if you want. Or combine it with my limiter on dust fraction used above.

**Reply by tilmes** [reply]
@cbardeen Do you have some code changes to test for me?  @andrewgettelman Yes, we could do some climatological analysis on the runs I have already done, and could add some dust reduction after that.

**Reply by tilmes** [reply]
@andrewgettelman I think, we decided that we are not adding additional changes, besides maybe removing the primary carbon model from BC. Have you looked into comparing this with observations or climatologies?

**Reply by andrewgettelman** [reply]
@tilmes , so you mean not anything in #3? Maybe we should chat. I'm going to talk about microphysics at the AMWG and could also talk about ice nucleation. I'm switching up the PUMAS code to put back in CNT immersion freezing with a 'factor' for dust (as we have already for BC) that will limit it to reasonable values. But we might want to propose some of the changes you have tested. Would be good to discuss before the AMWG. I'll be at FL next Thursday if we wanted to meet, or we could do it virtually.

**Comment by andrewgettelman** [comment]
@tilmes , I ran the latest updates to the microphysics with num_to_mass_in = .false.

With limfacdst = 0.05, there was almost no effect on the simulations (but it did something, not BFB)

I'm running again with limfacdst = 1.0 (no limiter on dust for ice nucleation) and num_to_mass_in =.false. and will report that next....(tomorrow)

**Comment by andrewgettelman** [comment]
Greetings @tilmes and @cbardeen. I did some testing as described. I did simulations with the following:

- Control CAM now (cam6357, cam_dev)
- Slightly modified control with different detrained ice size (Control Di-60)
- PUMAS with limfacdst = 0.05 (Dst0.05)
- PUMAS with limfacdst = 1 (Dst1). : This is the full immersion freezing
- PUMAS with limfacdst = 0.05 and num_to_mass_in = .false. (PUMAS Mass0.05)
- PUMAS with limfacdst = 1 and num_to_mass_in = .false. (PUMAS Mass1)

 For some reason I cannot attach the images here to Git hub, but I will send them

The results are that num_to_mass_in does quite a bit to reduce ice number and mass, but probably not enough. Ice number is still too high in the N. Hemisphere mid-latitudes, and a dust limiter is probably still necessary. 

So I will flip the switch for num_to_mass_in, and redo my runs to include it. The paper can then be used for documentation if you want. I can add Simone as a co-author if you want as well, and we can add perhaps a line or two about the change.

**Reply by tilmes** [reply]
@andrewgettelman Francis and I tested the code with all the changes in test 5 and compared to his new refactored code. We decided to set the limiter for Black Carbon to zero and get scientifically the same results.  The reason for proposing to set the BC limiter to zero is 1. it is already set to a very small value 0.01, second, in earlier discussions, we don't have a lot of confidence in what BC would do, third, the original CAM code is not considering the right radius for the primary mode and there are inconsistencies. 

@adamrher could you make an annual plot like you did before comparing these two cases to be able to compare to the earlier code? If they are fairly similar, I think, we can go ahead and to a pull request on Francis code. Maybe @andrewgettelman you can let us know what tuning factor we would need to apply after we have the plots?
This is the one I ran based on test5 without BC:
/glade/scratch/tilmes/archive/f.e22.Fnudged.f09_f09_mg17.cam6_3_082.2018.immfrz.005_no_bc
and Francis refactored code:
/glade/scratch/fvitt/archive/f.e22.Fnudged.f09_f09_mg17.cam6_3_082.2018.immfrz.005c

**Reply by tilmes** [reply]
@andrewgettelman I don't need to be co-author on your paper

**Reply by adamrher** [reply]
@tilmes the last run from Francis doesn't have PS in the output and so my script won't work for that run. But I added your 005.no_bc run in the lower right here:

![temp_dzonal_NUMICE](https://user-images.githubusercontent.com/29961476/214613670-425628a8-4b6a-4553-b775-1f6aa9897bd4.png)

**Reply by fvitt** [reply]
@adamrher I rerun my test case with more output including PS.  See history files in:
```
/glade/scratch/fvitt/archive/f.e22.Fnudged.f09_f09_mg17.cam6_3_082.2018.immfrz.005c/atm/hist
```

**Reply by adamrher** [reply]
@fvitt your run made a big difference to the ice number (plot on the bottom):

![temp_dzonal_NUMICE](https://user-images.githubusercontent.com/29961476/216487353-6ce15da3-404e-4b07-9d6b-60b9874a56d8.png)

**Reply by fvitt** [reply]
My immfrz.005c test case inadvertently had dust scaling factor set low:
```
 hetfrz_dust_scalfac            = 0.05D0
```
Will run another case with this set to 1.0

**Comment by fvitt** [comment]
New results are in
```
/glade/scratch/fvitt/archive/f.e22.Fnudged.f09_f09_mg17.cam6_3_082.2018.immfrz.005d/atm/hist
```
@adamrher Please provide the corresponding plot and we'll see how these compare.

**Reply by adamrher** [reply]
Here ya go,

![temp_dzonal_NUMICE](https://user-images.githubusercontent.com/29961476/216638988-d715ac32-a2fb-42cb-a900-7710916e0a06.png)

**Comment by tilmes** [comment]
@fvitt @adamrher @andrewgettelman Thanks, so  Francis' refactored code gives the same answers, with bc turned off. I am suggesting to do a pull request that uses Francis code, and we can list all the changes we have made (test 5 plus bc not considered). If @andrewgettelman prefers, we can also add the tuning factor of 0.05 back to Francis' case and we end up with very similar values compared to the background case.

**Reply by andrewgettelman** [reply]
Thanks @tilmes , this looks good. For purposes of putting it into CAM, probably best to have the dust fraction at 0.05 to keep the results similar to the current version. Relaxing that number towards 1 could be done while tuning up CAM7.
